include/open62541/server.h

```diff
@@ -1886,6 +1891,22 @@ typedef struct {
 UA_ServerStatistics UA_EXPORT
 UA_Server_getStatistics(UA_Server *server);
 
+/* Reverse connect */
+typedef enum {
+    REVERSE_CONNECT_STATUS_INITIAL = 0,
+    REVERSE_CONNECT_STATUS_CONNECTING,
+    REVERSE_CONNECT_STATUS_CONNECTED,
+    REVERSE_CONNECT_STATUS_DISCONNECTED
```

Comment:
What happens if the TCP connection opens, but the connection SecureChannel fails to be created.
For example due to a problem with certificates.
Will there still be a _CONNECTED or just _DISCONNECTED?

It should be documented how the callback relates to the SecureChannel state.

Reply:
I'd say it's just a DISCONNECTED. The user can't really do anything about it on the client side, so I see no point in setting the state to CONNECTED. But we could split the CONNECTED state into two states (connection has been opened vs. secure channel is up). This could permit debugging connection issues, for example firewall problems vs. wrong security configuration in client or server.

Reply:
Debugging should be possible from the log output.
Having just _connected works for me as well.
But it should be clearly documented what _connected means.

we should use _disconnected for fresh reverse connections as well. See above.